app/code/Magento/CatalogGraphQl/DataProvider/Product/LayeredNavigation/AttributeOptionProvider.php

```diff
@@ -42,15 +42,17 @@ public function __construct(ResourceConnection $resourceConnection)
      *
      * @param array $optionIds
      * @param array $attributeCodes
+     * @param int|null $storeId
      * @return array
      * @throws \Zend_Db_Statement_Exception
      */
-    public function getOptions(array $optionIds, array $attributeCodes = []): array
```

Comment:
What is the reason of adding parameter (optional) before other optional parameter

Reply:
Correct me if I am wrong, but in our case Store ID is not optional but required. Optional is only $attributeCodes. This is why I added Store ID before it.

Reply:
The question mark before type means that you allow null there

Reply:
Yes, but as I know it is not optional, we have to pass int or null (correct me if I am wrong).

I am passing it with a question mark because the method I am calling it from also has it in params with a question mark. 

```
/**
     * Get list of attributes with options
     *
     * @param AggregationInterface $aggregation
     * @param int|null $storeId
     * @return array
     * @throws \Zend_Db_Statement_Exception
     */
    private function getAttributeOptions(AggregationInterface $aggregation, ?int $storeId): array
    {
        $attributeOptionIds = [];
        $attributes = [];
        foreach ($this->getAttributeBuckets($aggregation) as $bucket) {
            $attributes[] = \preg_replace('~_bucket$~', '', $bucket->getName());
            $attributeOptionIds[] = \array_map(
                function (AggregationValueInterface $value) {
                    return $value->getValue();
                },
                $bucket->getValues()
            );
        }

        if (!$attributeOptionIds) {
            return [];
        }

        return $this->attributeOptionProvider->getOptions(\array_merge(...$attributeOptionIds), $storeId, $attributes);
    }
```

Reply:
Now I understand. Thank you for your effort to explain that!

Reply:
No problem, thanks!

Reply:
@lbajsarowicz could you mark this as resolved, please? It is blocking us to merge this fix.